client/extensions/woocommerce/state/data-layer/products/index.js

```diff
@@ -75,3 +84,46 @@ export function handleProductRequest( { dispatch }, action ) {
 	const updatedSuccessAction = updatedAction( siteId, action, successAction );
 	dispatch( get( siteId, 'products/' + productId, updatedSuccessAction, failureAction ) );
 }
+
+export function handleProductsRequestSuccess( store, action, next, response ) {
+	const { siteId, page } = action;
+	const { headers, body, status } = response.data;
+
+	// We needed to envelope the reponse to get the number of products and pages. see handleProductsRequest
+	if ( status !== 200 ) {
```

Comment:
Why would this ever get back something other than a 200 and fire `handleProductsRequestSuccess`? If it's not successful, shouldn't it fire `handleProductsRequestError` instead? Or is another 2XX code possible that you're checking for?